### DISCUSSION TITLE
Failed during configuration: Have not found any log file for custom-postgresql jai

### DISCUSSION THREAD

**Discussion by mdpstreet** [fail2ban/fail2ban#3424] [General]
Hi All,

I am fairly new to Fail2Ban and I cannot get a custom jail for Postgres to read the log file no matter what name I call the log file. 

![image](https://user-images.githubusercontent.com/118403453/204991532-c3ee351a-ecf6-47d0-9746-d1a7689f1644.png)

Below is my jail file I have configured

![image](https://user-images.githubusercontent.com/118403453/204992499-8e37904e-eb98-4698-abf8-2f10dada3249.png)

Below is the Custom filter file.

![image](https://user-images.githubusercontent.com/118403453/204992870-2d477103-65b0-4f6e-afaf-73f44a255222.png)

Below is the path showing that the postgresql.log exists.

![image](https://user-images.githubusercontent.com/118403453/204993203-c640fc6f-5545-4ff2-8cff-9ccdff64764e.png)

I have enabled the sshd jail and it is running fine. I have to disable the custom Postgresql jail to get to start the fail2ban service.
I have done multiple searches trying to find anyone having the same issues but could not find any. My OS is AlmaLinux 9 with PostgreSQL 15

Any help would be appreciated.

Regards,

Mark Street

**Comment by sebres** [comment]
If it is not a typo in `logpath`, it may be some permissions issue. 
1. SELinux? If so check the permissions (`semanage` and co), remove some hardening rules (like #2162) or white-list paths via `ReadWritePaths` specified in fail2ban.service (or fail2ban.service.d/override.conf). 
I guess path `/data/PostgreSQL/...` cannot be accessed from service fail2ban is running in.
Also to verify the issue, you can try to start fail2ban from your console:
```bash
systemctl stop fail2ban; # stop fail2ban service (systemd unit)
fail2ban-server --logtarget=stdout -f start; # start f2b-server in foreground mode (ctrl+c to stop)
```
(if it is starting there, you have permissions issue starting it as a service)

2. If it is running in a container (docker, kubernetes, etc) you need to bind mount, file or directory of host (or container) where PostgreSQL is logging,  and set proper `logpath`.

3. You could configure PostgreSQL to log to systemd journal (check whether you'd see its messages with `journalctl -u postgresql`) and then configure the jail to monitor journal instead of log file:
```ini
[custom-postgresql]
backend = systemd
...
enabled = true
```

**Comment by mdpstreet** [comment]
Thanks so much for the advice. Totally forgotten about SELinux. I am sure this is the problem. Will test it out and let you know

**Comment by mdpstreet** [comment]
Hi Sergey,

Thanks so much for you help. Yes it was SELinux. I have turned it off until I finish my testing and then I will work out the SELinux policy. 

Regards,

Mark Street
